openstack/check.py

```diff
@@ -850,7 +852,21 @@ def get_scoped_project(self, instance):
         """
         project_auth_scope = self.get_scope_for_instance(instance)
         if project_auth_scope.tenant_id:
-            return {"id": project_auth_scope.tenant_id}
+            scope = {"id": project_auth_scope.tenant_id}
+            if project_auth_scope.project_name:
+                scope["name"] = project_auth_scope.project_name
+            else:
+                # Try to get the project name from the keystone API so that it may be added as a tag
+                url = "{0}/{1}/{2}/{3}".format(self.keystone_server_url, DEFAULT_KEYSTONE_API_VERSION, "projects", project_auth_scope.tenant_id)
+                headers = {'X-Auth-Token': self.get_auth_token(instance)}
+                try:
+                    project_details = self._make_request_with_auth_fallback(url, headers)
+                    # Set the project name so we won't have to fetch it next time
+                    project_auth_scope.project_name = project_details["project"]["name"]
```

Comment:
Why not just add the `project_name` when we make the query the first time? I don't think it makes sense to query the projects endpoint yet another time to collect the `project_name` when it should already be provided int the first query (this is the call we're making: https://developer.openstack.org/api-ref/identity/v3/index.html#list-projects). We could just do that here if I'm not mistaken (with a single line of code): https://github.com/DataDog/integrations-core/pull/515/files#diff-3731b2d63840c0acb3cc77f20893b8b9L868